infra/modules/providers/azure/storage-account/main.tf

```diff
@@ -0,0 +1,32 @@
+resource "azurerm_storage_account" "sa" {
+  resource_group_name      = var.resource_group_name
+  location                 = var.resource_group_location
+  name                     = lower(var.account_name)
+  account_tier             = var.performance_tier
+  account_replication_type = var.replication_type
+
+  # optional
+  account_kind              = var.kind
+  enable_https_traffic_only = var.https
+  account_encryption_source = var.encryption_source
+
+  # enrolls storage account into azure 'managed identities' authentication
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+resource "azurerm_storage_container" "sa" {
+  name                  = var.storage_container_name
+  resource_group_name   = var.resource_group_name
+  storage_account_name  = azurerm_storage_account.sa.name
+  container_access_type = "private"
+}
+
+module "service-principal" {
+  source       = "../service-principal"
+  object_id    = var.existing_sp_object_id
+  display_name = ""
```

Comment:
Why `""`? Unclear on where this will pop up in the UI, but I can see having a display name being useful for tracking

Reply:
This has been filed as a bug within our repo. #182  The "display name" var isn't used when taking the "existing service principal" path (the path my module has chosen). 

The "display name" var should only be required and accessed during "new service principal" scenarios. In other words, we desire that this variable be 'null' and not evaluated for "existing service principal" scenarios, however, it's evaluated even when null which causes an error. In order to avoid the error, i have to give it a value.

Reply:
oh interesting! thanks for pointing this out. Another option is to fix the bug by adding a default variable, seems like it would be simple to do and would drive down tech debt for our project.

Your call.

Reply:
Bug fix has been applied and merged into master.